<launch>
  <!-- Launch AruCo Detect Qr code stuff -->
  <arg name="camera_num" default="2"/>
  <include file="$(find windblade_scanner)/launch/2_robot_setup/turtlebot/world_frame/asus_xtion_pro/asus_xtion_pro_aruco_detect_qr$(arg camera_num).launch"/>
  
  <arg name="robot" default="turtlebot"/>
  <arg name="save_file_name" default="experiment_1_results.csv"/>
  <!-- Directory to save the data from the experiment -->
  <arg name="experiment_save_directory" default="$(find windblade_scanner)/experimental_data/$(arg robot)/$(arg save_file_name)"/>
  <!-- Topic Parameters -->
  <arg name="odom_topic" default="/odom"/>
  <arg name="fiducial_tf_topic" default="/fiducial_transforms"/>
  <arg name="cmd_vel_topic" default="/mobile_base/commands/velocity"/>
  <!-- Experiment Parameters -->
  <arg name="waypoint_ang_tol" default="0.05"/>
  <arg name="min_frames" default="60"/>
  <arg name="dist_meas_min" default="0.3"/>
  <arg name="dist_meas_max" default="1.6"/>
  <arg name="dist_meas_step" default="11"/>
  <arg name="aruco_code" default="/fiducial_0"/>

  <!-- Launches TF publisher for world frame. -->
  <node pkg="windblade_scanner" type="world_frame_publisher_experiment_1.py" name="world_frame_publisher" output="screen" >
    <!-- Directory to save the data from the experiment -->
    <param name="experiment_save_directory" value="$(arg experiment_save_directory)"/>
    <!-- Topic Parameters -->
    <param name="odom_topic" value="$(arg odom_topic)"/>
    <param name="fiducial_tf_topic" value="$(arg fiducial_tf_topic)"/>
    <param name="cmd_vel_topic" value="$(arg cmd_vel_topic)"/>
    <!-- Experiment Parameters -->
    <param name="waypoint_ang_tol" value="$(arg waypoint_ang_tol)"/>
    <param name="min_frames" value="$(arg min_frames)"/>
    <param name="dist_meas_min" value="$(arg dist_meas_min)"/>
    <param name="dist_meas_max" value="$(arg dist_meas_max)"/>
    <param name="dist_meas_step" value="$(arg dist_meas_step)"/>
    <param name="aruco_code" value="$(arg aruco_code)"/>
  </node>

</launch>